File: src/common/py/core/messaging/meta/command_reply_meta_information.py
import dataclasses

from .message_meta_information import MessageMetaInformation


@dataclasses.dataclass(frozen=True, kw_only=True)
class CommandReplyMetaInformation(MessageMetaInformation):
    """
    Message meta information specific to ``CommandReply``.
    """

    _is_handled_externally: bool = False

    @property
    def is_handled_externally(self) -> bool:
        """
        Whether the message is handled outside the message bus.
        """
        return self._is_handled_externally

    def set_handled_externally(self, value: bool = True) -> None:
        """
        Sets whether the message is handled outside the message bus.
        """
        object.__setattr__(self, "_is_handled_externally", value)

File: src/common/web/core/messaging/meta/CommandReplyMetaInformation.ts
import { MessageMetaInformation } from "./MessageMetaInformation";

/**
 * Message meta information specific to ``CommandReply``.
 */
export class CommandReplyMetaInformation extends MessageMetaInformation {
    private _isHandledExternally: boolean = false;

    /**
     * Whether the message is handled outside the message bus.
     */
    public get isHandledExternally(): boolean {
        return this._isHandledExternally;
    }

    /**
     * Sets whether the message is handled outside the message bus.
     */
    public set isHandledExternally(handled: boolean) {
        this._isHandledExternally = handled;
    }
}
